### PR TITLE
[Dubbo-3727] Fix NullPointerException when protocoluff protocol serializes an empty object

### LINKED ISSUE #3727: 使用protostuff序列化event request未对data进行空安全检查
:: Dubbo Spring Boot (v2.7.0) : https://github.com/apache/incubator-dubbo-spring-boot-project
 :: Dubbo (v2.7.1-SNAPSHOT) : https://github.com/apache/incubator-dubbo
 :: Discuss group : dev@dubbo.apache.org
由于心跳检查发送的request的mData属性为空，
而在encodeEventData时为判断是否为空导致的空指针异常
image
image
image

### PR DESCRIPTION
**Pull request by zhangfei9734**
## What is the purpose of the change

Fixed the problem that the protocoluff protocol is not judged empty when serialized

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-serialization/dubbo-serialization-test/src/test/java/org/apache/dubbo/common/serialize/protostuff/ProtostuffObjectOutputTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ProtostuffObjectOutputTest {
```
Comment (CrazyHZM) [inline dubbo-serialization/dubbo-serialization-test/src/test/java/org/apache/dubbo/common/serialize/protostuff/ProtostuffObjectOutputTest.java]: Dubbo has a dubbo-serialization-test module that can be added to the module.

File: dubbo-serialization/dubbo-serialization-test/src/test/java/org/apache/dubbo/common/serialize/protostuff/ProtostuffObjectOutputTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package org.apache.dubbo.common.serialize.protostuff;
```
Comment (kexianjun) [inline dubbo-serialization/dubbo-serialization-test/src/test/java/org/apache/dubbo/common/serialize/protostuff/ProtostuffObjectOutputTest.java]: a license header is needed here

**Comment by CrazyHZM** [comment]
LGTM

**Comment by zhangfei9734** [comment]
@CrazyHZM Will this merge request be released in 2.7.1-RELEASE?

**Comment by CrazyHZM** [comment]
> @CrazyHZM Will this merge request be released in 2.7.1-RELEASE?

 This merge request will be released in 2.7.2.

**Pull request merged**